data/src/main/java/com/ronnie/data/mediator/PixaRemoteMediator.kt

```diff
@@ -9,95 +9,95 @@ import com.ronnie.commons.FIRST_PAGE
-            }
-            LoadType.PREPEND -> {
-                val remoteKeys = getRemoteKeyForFirstItem(state)
-                val prevKey = remoteKeys?.prevPage
-                    ?: return MediatorResult.Success(endOfPaginationReached = remoteKeys != null)
-                prevKey
-            }
-            LoadType.APPEND -> {
-                val remoteKeys = getRemoteKeyForLastItem(state)
-                val nextKey = remoteKeys?.nextPage
-                    ?: return MediatorResult.Success(endOfPaginationReached = remoteKeys != null)
-                nextKey
-            }
-        }
-
-        try {
-            val response = pixaBayApi.searchImages(searchString, state.config.pageSize, page)
-            val images = response.images
-
-            images.map {
-                it.searchTerm = searchString
-            }
-
-            val endOfPaginationReached = images.isEmpty()
-            db.withTransaction {
-                if (loadType == LoadType.REFRESH) {
-                    db.imageDao().clearAll()
-                    db.remoteKeyDao().clearRemoteKeys()
-                }
-                val prevKey = if (page == FIRST_PAGE) null else page - 1
-                val nextKey = if (endOfPaginationReached) null else page + 1
-
-                val keys = images.map {
-                    RemoteKey(prevPage = prevKey, nextPage = nextKey, imageId = it.id)
-                }
-                db.remoteKeyDao().insertAll(keys)
-                db.imageDao().insertAll(images)
-            }
-
-            return MediatorResult.Success(endOfPaginationReached = endOfPaginationReached)
-        } catch (exception: IOException) {
-            return MediatorResult.Error(exception)
-        } catch (exception: HttpException) {
-            return MediatorResult.Error(exception)
-        }
-
-    }
-
-    override suspend fun initialize(): InitializeAction {
-        return InitializeAction.LAUNCH_INITIAL_REFRESH
-    }
-
-    private suspend fun getRemoteKeyForLastItem(state: PagingState<Int, Image>): RemoteKey? {
-        return state.pages.lastOrNull { it.data.isNotEmpty() }?.data?.lastOrNull()
-            ?.let { image ->
-                db.remoteKeyDao().remoteKeysImageId(image.id)
-            }
-    }
-
-    private suspend fun getRemoteKeyForFirstItem(state: PagingState<Int, Image>): RemoteKey? {
-        return state.pages.firstOrNull { it.data.isNotEmpty() }?.data?.firstOrNull()
-            ?.let { image ->
-                db.remoteKeyDao().remoteKeysImageId(image.id)
-            }
-    }
-
-    private suspend fun getRemoteKeyClosestToCurrentPosition(
-        state: PagingState<Int, Image>
-    ): RemoteKey? {
-        return state.anchorPosition?.let { position ->
-            state.closestItemToPosition(position)?.let { image ->
-                db.remoteKeyDao().remoteKeysImageId(image.id)
-            }
-        }
-    }
-}
\ No newline at end of file
+//
```

Comment:
Wouldnt it be nice if you just remove the code rather than commenting?

Reply:
ok